lib/Notification/Listener.php

```diff
@@ -273,17 +279,52 @@ protected function sendCallNotifications(Room $room): void {
 			return;
 		}
 
+		$this->preparedCallNotifications = [];
 		$userIds = $this->participantsService->getParticipantUserIdsForCallNotifications($room);
+		// Room name depends on the notification user for one-to-one,
+		// so we avoid pre-parsing it there. Also, it comes with some base load,
+		// so we only do it for "big enough" calls.
+		$preparseNotificationForPush = count($userIds) > 10;
+		if ($preparseNotificationForPush) {
+			$fallbackLang = $this->serverConfig->getSystemValue('force_language', null);
+			if (is_string($fallbackLang)) {
+				/** @psalm-var array<string, string> $userLanguages */
+				$userLanguages = [];
+			} else {
+				$fallbackLang = $this->serverConfig->getSystemValueString('default_language', 'en');
+				/** @psalm-var array<string, string> $userLanguages */
+				$userLanguages = $this->serverConfig->getUserValueForUsers('core', 'lang', $userIds);
+			}
+		}
+
 		$this->connection->beginTransaction();
 		try {
 			foreach ($userIds as $userId) {
 				if ($actorId === $userId) {
 					continue;
 				}
 
+				if ($preparseNotificationForPush) {
+					// Get the settings for this particular user, then check if we have notifications to email them
+					$languageCode = $userLanguages[$userId] ?? $fallbackLang;
+
+					if (!isset($this->preparedCallNotifications[$languageCode])) {
+						$translatedNotification = clone $notification;
+
+						$this->notificationManager->setPreparingPushNotification(true);
+						$this->preparedCallNotifications[$languageCode] = $this->notificationProvider->prepare($translatedNotification, $languageCode);
+						$this->notificationManager->setPreparingPushNotification(false);
+						$userNotification = $translatedNotification;
+					} else {
+						$userNotification = $this->preparedCallNotifications[$languageCode];
+					}
```

Comment:
Would it be the same? 
```suggestion
					}
					$userNotification = $this->preparedCallNotifications[$languageCode];
```

Reply:
Right, fixed